airflow/utils/timeout.py

```diff
@@ -39,8 +40,9 @@ def handle_timeout(self, signum, frame):
 
     def __enter__(self):
         try:
-            signal.signal(signal.SIGALRM, self.handle_timeout)
-            signal.alarm(self.seconds)
+            if isinstance(threading.current_thread(), threading._MainThread):
```

Comment:
We'd have to make `process_file` stop using timeout if it's not on the main thread (https://github.com/apache/incubator-airflow/blob/master/airflow/models.py#L252)